<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LoginDialog</class>
 <widget class="QDialog" name="LoginDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>350</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string notr="true">Login</string>
  </property>
  <layout class="QVBoxLayout">
   <item>
    <widget class="QLabel" name="serverTitle">
     <property name="text">
      <string notr="true">&lt;i&gt;Connecting...&lt;/i&gt;</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="openExternalLinks">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QStackedWidget" name="pages">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="loadingPage">
      <layout class="QVBoxLayout">
       <item>
        <widget class="widgets::Spinner" name="loadingSpinner" native="true">
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>64</height>
          </size>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="authPage">
      <layout class="QVBoxLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="loginPromptLabel">
         <property name="styleSheet">
          <string notr="true">background: #4d4d4d;
color: #fcfcfc;
padding: 16px;</string>
         </property>
         <property name="text">
          <string notr="true">LOG IN</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QGridLayout" name="authPageLayout">
         <item row="1" column="2">
          <widget class="QLineEdit" name="username">
           <property name="placeholderText">
            <string>Username</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QLabel" name="passwordIcon">
           <property name="text">
            <string notr="true">P</string>
           </property>
          </widget>
         </item>
         <item row="3" column="2">
          <widget class="QLabel" name="badPasswordLabel">
           <property name="styleSheet">
            <string notr="true">color: red</string>
           </property>
           <property name="text">
            <string>Incorrect password.</string>
           </property>
          </widget>
         </item>
         <item row="4" column="2">
          <widget class="QCheckBox" name="rememberPassword">
           <property name="text">
            <string>Remember my password</string>
           </property>
          </widget>
         </item>
         <item row="2" column="3">
          <spacer name="authPageRightSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="2" column="0">
          <spacer name="authPageLeftSpacer">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>0</width>
             <height>0</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="2" column="2">
          <widget class="QLineEdit" name="password">
           <property name="echoMode">
            <enum>QLineEdit::Password</enum>
           </property>
           <property name="placeholderText">
            <string>Password</string>
           </property>
           <property name="mandatoryfield" stdset="0">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLabel" name="usernameIcon">
           <property name="text">
            <string notr="true">U</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QComboBox" name="avatarList">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>42</height>
            </size>
           </property>
           <property name="iconSize">
            <size>
             <width>32</width>
             <height>32</height>
            </size>
           </property>
           <property name="frame">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="authPageFixedSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>6</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QLabel" name="authExplanationLabel">
         <property name="text">
          <string notr="true">&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Login explanation goes here.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="textFormat">
          <enum>Qt::RichText</enum>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
         <property name="openExternalLinks">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="authPageBottomSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="listingPage">
      <layout class="QVBoxLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="listingPageFilterLayout">
         <item>
          <widget class="QLineEdit" name="filter">
           <property name="placeholderText">
            <string>Filter</string>
           </property>
           <property name="clearButtonEnabled">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QToolButton" name="showNsfw">
           <property name="toolTip">
            <string>Show sessions not suitable for minors (NSFM)</string>
           </property>
           <property name="icon">
            <iconset>
             <normaloff>:/icons/censored.svg</normaloff>:/icons/censored.svg</iconset>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTableView" name="sessionList">
         <property name="selectionMode">
          <enum>QAbstractItemView::SingleSelection</enum>
         </property>
         <property name="selectionBehavior">
          <enum>QAbstractItemView::SelectRows</enum>
         </property>
         <property name="gridStyle">
          <enum>Qt::NoPen</enum>
         </property>
         <property name="sortingEnabled">
          <bool>true</bool>
         </property>
         <attribute name="horizontalHeaderHighlightSections">
          <bool>false</bool>
         </attribute>
         <attribute name="verticalHeaderVisible">
          <bool>false</bool>
         </attribute>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="sessionPasswordPage">
      <layout class="QVBoxLayout">
       <item>
        <widget class="QLabel" name="sessionPasswordMessageLabel">
         <property name="text">
          <string>This session is password-protected.</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="sessionPasswordTopSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QHBoxLayout" name="sessionPasswordLayout">
         <item>
          <widget class="QLabel" name="sessionPasswordLabel">
           <property name="text">
            <string>Password:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="sessionPassword">
           <property name="echoMode">
            <enum>QLineEdit::Password</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QLabel" name="badSessionPasswordLabel">
         <property name="styleSheet">
          <string notr="true">color:red;</string>
         </property>
         <property name="text">
          <string>Incorrect password.</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="sessionPasswordNoticeLabel">
         <property name="text">
          <string>Enter the &lt;b&gt;session password&lt;/b&gt;, &lt;i&gt;not&lt;/i&gt; your account password.</string>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="sessionPasswordPageBottomSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="catchupPage">
      <layout class="QVBoxLayout" stretch="0,0,0">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="catchupMessageLabel">
         <property name="text">
          <string>Catching up...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QProgressBar" name="progressBar">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>0</number>
         </property>
         <property name="textVisible">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="catchupBottomSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>58</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="certChangedPage">
      <layout class="QGridLayout" name="certChangedGridLayout" columnstretch="0,0,0">
       <item row="1" column="1">
        <spacer name="certChangedPageTopSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="viewOldCert">
         <property name="text">
          <string>View old</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="viewNewCert">
         <property name="text">
          <string>View new</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1" colspan="2">
        <widget class="QCheckBox" name="replaceCert">
         <property name="text">
          <string>Replace pinned certificate</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1" colspan="2">
        <widget class="QLabel" name="errorMessage">
         <property name="text">
          <string notr="true">Server SSL certificate has changed!</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <spacer name="certChangedPageBottomSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>10</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="0" rowspan="4">
        <widget class="QLabel" name="warningIcon">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string notr="true">ERROR</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="nsfmConfirmPage">
      <layout class="QVBoxLayout" name="nsfmConfirmLayout">
       <item>
        <widget class="QLabel" name="nsfmConfirmTitle">
         <property name="textFormat">
          <enum>Qt::RichText</enum>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="nsfmConfirmLabel">
         <property name="text">
          <string>&lt;p&gt;This session has indicated that its contents are &lt;em&gt;not suitable for minors&lt;/em&gt; (NSFM).&lt;/p&gt;&lt;p&gt;&lt;strong&gt;You must be over the age of 18 and be allowed to consume adult content to proceed.&lt;/strong&gt;&lt;/p&gt;&lt;p&gt;Are you sure you want to join this session?&lt;/p&gt;</string>
         </property>
         <property name="textFormat">
          <enum>Qt::RichText</enum>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="nsfmConfirmSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QCheckBox" name="nsfmConfirmAgainBox">
         <property name="text">
          <string>Show this warning again next time</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="authModePage">
      <layout class="QGridLayout" name="authModePageLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="0" column="0" colspan="3">
        <widget class="QLabel" name="authModePromptLabel">
         <property name="styleSheet">
          <string notr="true">background: #4d4d4d;
color: #fcfcfc;
padding: 16px;</string>
         </property>
         <property name="text">
          <string notr="true">LOG IN</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="0" rowspan="4">
        <spacer name="authModePageLeftSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="2" column="2" rowspan="4">
        <spacer name="authModePageRightSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="4" column="1">
        <widget class="QPushButton" name="methodGuestButton">
         <property name="text">
          <string>Continue without account</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0" colspan="3">
        <widget class="QLabel" name="methodExplanationLabel">
         <property name="text">
          <string notr="true">Explanation text goes here.</string>
         </property>
         <property name="textFormat">
          <enum>Qt::RichText</enum>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
         <property name="openExternalLinks">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="8" column="0" colspan="3">
        <spacer name="authModePageBottomSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="1">
        <widget class="QPushButton" name="methodAuthButton">
         <property name="text">
          <string>Log in with server account</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="methodExtAuthButton">
         <property name="text">
          <string notr="true">Log in with extauth account</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0" colspan="3">
        <spacer name="authModePageFixedSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Fixed</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>6</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="recentPage">
      <layout class="QGridLayout" name="recentGridLayout">
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item row="11" column="0" colspan="3">
        <spacer name="recentBottomSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="0" colspan="3">
        <widget class="QLabel" name="recentPromptLabel">
         <property name="styleSheet">
          <string notr="true">background: #4d4d4d;
color: #fcfcfc;
padding: 16px;</string>
         </property>
         <property name="text">
          <string notr="true">LOG IN</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="2" column="0" rowspan="2">
        <spacer name="recentLeftSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="6" column="1">
        <widget class="QPushButton" name="recentEditButton">
         <property name="toolTip">
          <string>Continue with this account, but change the avatar, username or password.</string>
         </property>
         <property name="text">
          <string>Edit and log in</string>
         </property>
         <property name="icon">
          <iconset theme="edit-rename">
           <normaloff>theme:edit-rename.svg</normaloff>theme:edit-rename.svg</iconset>
         </property>
        </widget>
       </item>
       <item row="10" column="1">
        <widget class="QPushButton" name="recentBreakButton">
         <property name="toolTip">
          <string>Don't use any of these accounts.</string>
         </property>
         <property name="text">
          <string>Log in a different way</string>
         </property>
         <property name="icon">
          <iconset theme="application-menu">
           <normaloff>theme:application-menu.svg</normaloff>theme:application-menu.svg</iconset>
         </property>
        </widget>
       </item>
       <item row="8" column="0" colspan="3">
        <spacer name="recentTopSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>6</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="1">
        <widget class="QComboBox" name="recentAccountCombo">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>42</height>
          </size>
         </property>
         <property name="iconSize">
          <size>
           <width>32</width>
           <height>32</height>
          </size>
         </property>
         <property name="frame">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QPushButton" name="recentLogInButton">
         <property name="toolTip">
          <string>Log in with the selected account.</string>
         </property>
         <property name="text">
          <string>Log in with this account</string>
         </property>
         <property name="icon">
          <iconset theme="checkbox">
           <normaloff>theme:checkbox.svg</normaloff>theme:checkbox.svg</iconset>
         </property>
        </widget>
       </item>
       <item row="2" column="2" rowspan="2">
        <spacer name="recentRightSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="7" column="1">
        <widget class="QPushButton" name="recentForgetButton">
         <property name="toolTip">
          <string>Forget this account.</string>
         </property>
         <property name="text">
          <string>Forget this account</string>
         </property>
         <property name="icon">
          <iconset theme="trash-empty">
           <normaloff>theme:trash-empty.svg</normaloff>theme:trash-empty.svg</iconset>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="rulesPage">
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QTextBrowser" name="rulesTextBrowser">
         <property name="openExternalLinks">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="rulesRememberBox">
         <property name="text">
          <string>Don't ask again unless the rules change</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::No|QDialogButtonBox::Ok|QDialogButtonBox::Yes</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>widgets::Spinner</class>
   <extends>QWidget</extends>
   <header>desktop/widgets/spinner.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>avatarList</tabstop>
  <tabstop>username</tabstop>
  <tabstop>password</tabstop>
  <tabstop>rememberPassword</tabstop>
  <tabstop>filter</tabstop>
  <tabstop>showNsfw</tabstop>
  <tabstop>sessionList</tabstop>
  <tabstop>sessionPassword</tabstop>
  <tabstop>viewOldCert</tabstop>
  <tabstop>viewNewCert</tabstop>
  <tabstop>replaceCert</tabstop>
  <tabstop>nsfmConfirmAgainBox</tabstop>
  <tabstop>methodExtAuthButton</tabstop>
  <tabstop>methodAuthButton</tabstop>
  <tabstop>methodGuestButton</tabstop>
  <tabstop>recentAccountCombo</tabstop>
  <tabstop>recentLogInButton</tabstop>
  <tabstop>recentEditButton</tabstop>
  <tabstop>recentForgetButton</tabstop>
  <tabstop>recentBreakButton</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
